multi-connector/src/main/scala/it/agilelab/darwin/connector/multi/MultiConnector.scala

```diff
@@ -0,0 +1,324 @@
+package it.agilelab.darwin.connector.multi
+
+import it.agilelab.darwin.common.Connector
+import it.agilelab.darwin.common.compat.RightBiasedEither
+import it.agilelab.darwin.manager.SchemaPayloadPair
+import it.agilelab.darwin.manager.exception.DarwinException
+import it.agilelab.darwin.manager.util.{ AvroSingleObjectEncodingUtils, ConfluentSingleObjectEncoding }
+import org.apache.avro.Schema
+
+import java.io.{ ByteArrayInputStream, InputStream, OutputStream, SequenceInputStream }
+import java.nio.{ ByteBuffer, ByteOrder }
+
+class MultiConnector(
+  val registrator: Connector,
+  val confluentConnectors: List[Connector],
+  val singleObjectEncodingConnectors: List[Connector]
+) extends Connector {
+
+  override def createTable(): Unit = registrator.createTable()
+
+  /**
+    * Returns whether or not the configured table exists
+    */
+  override def tableExists(): Boolean = registrator.tableExists()
+
+  /**
+    * @return a message indicating the user what he/she should do to create the table him/herself
+    */
+  override def tableCreationHint(): String = registrator.tableCreationHint()
+
+  /**
+    * Loads all schemas found on the storage.
+    * This method can be invoked multiple times: to initialize the initial values or to update the existing ones with
+    * the new data found on the storage.
+    *
+    * @return a sequence of all the pairs (ID, schema) found on the storage
+    */
+  override def fullLoad(): Seq[(Long, Schema)] =
+    confluentConnectors.flatMap(_.fullLoad()) ++ singleObjectEncodingConnectors.flatMap(_.fullLoad())
```

Comment:
I think we should specify here that in this implementation it could happen that we have two entries for the same ID. It could happen also using a single connector, but in that case, it should be viewed as an error, while here it could mean that a schema is mapped into two or more connectors. I think that if we have the same schema in both systems with the same ID we should return it only once.

Reply:
Collisions is a very hot topic in this Multi connector case. actually this comment made me realise that it does not make any sense to allow multiple confluent-like connectors, because there are very high chance of collision with tremendous consequences. In the case of single-object-encoding (i.e. 10 bytes header using schema fingerprinting) the chance of same fingerprint and different schema is so low, that I would not bother at all (as we do in all the codebase) therefore a simple "distinct" on this collection might be enough, what do you think?

Reply:
I thought that if a user is using two confluent connectors he/she must be sure that both of them contain the same IDs for the same schemas (even if I don't know if this is something realistic).
If you think that you cannot enforce two confluent connectors to have the same IDs for the same schemas, then I agree with you, we should change it from a collection to a single confluent connector.

Another important thing: if schema X is mapped in a single-object-encoded (e.g. ID1->X) and a confluent registry (e.g. ID2->X) there should be no problem even if they have different IDs: when someone asks for ID1 or ID2 we will return X anyway. But maybe it could happen that we have the same ID for two schemas one in the single-object-encoded and one in the confluent one (even if one is 8 bytes and the other one is just 4, because the first 4 bytes of the long cold be all zeros). In that case a distinct could lead to wrong results. Am I right?

Reply:
> I thought that if a user is using two confluent connectors he/she must be sure that both of them contain the same IDs for the same schemas (even if I don't know if this is something realistic).
If you think that you cannot enforce two confluent connectors to have the same IDs for the same schemas, then I agree with you, we should change it from a collection to a single confluent connector.

I think that it is not possible to have two confluent schema registry perfectly aligned, if someone wants redundancy it should use multiple endpoints.

> In that case a distinct could lead to wrong results. Am I right?
I would do the distinct on the pair, not only on the ID, anyway, the problem of duplicate id, I think it is not possible to solve it.